lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -719,49 +722,43 @@ export default class BitcoinProcessor {
   }
 
   /**
-   * Return transactions grouped by block height. EX: [[transactions in block1], [transactions in block2], [transactions in block3]]
+   * Return transactions since transaction number and number of blocks acquired (Will get at least pageSizeInBlocks)
    * @param since Transaction number to query since
-   * @param numOfBlocks number of blocks worth of transactions to get at least
+   * @param currentLastProcessedBlock The current last processed block. Not using this.lastProcessedBlock to avoid value changing during processing
+   * @returns a tuple of [transactions, numberOfBlocksContainedInTransactions]
    */
-  private async getAtLeastPageSizeBlocksWorthOfTransactionsSinceTransactionNumber (since: number | undefined): Promise<TransactionModel[][]> {
-    let beginTransactionNumber = since === undefined ? 0 : TransactionNumber.getBlockNumber(since);
-    let endTransactionNumber = beginTransactionNumber + BitcoinProcessor.pageSizeInBlocks;
+  private async getTransactionsSince (since: number | undefined, currentLastProcessedBlock: IBlockInfo): Promise<[TransactionModel[], number]> {
+    let beginTransactionTime = since === undefined ? this.genesisBlockNumber : TransactionNumber.getBlockNumber(since);
+    let numOfBlocksAcquired = 0;
 
-    const groupedTransactionsToReturn: TransactionModel[][] = [];
+    const transactionsToReturn: TransactionModel[] = [];
 
     // while need more blocks and have not reached the processed block
-    while (groupedTransactionsToReturn.length < BitcoinProcessor.pageSizeInBlocks && beginTransactionNumber <= this.lastProcessedBlock!.height) {
-      let transactions: TransactionModel[] = await this.transactionStore.getTransactionsByTransactionTime(beginTransactionNumber, endTransactionNumber);
+    while (numOfBlocksAcquired < BitcoinProcessor.pageSizeInBlocks && beginTransactionTime <= currentLastProcessedBlock.height) {
+      const endTransactionTime = beginTransactionTime + BitcoinProcessor.pageSizeInBlocks;
```

Comment:
Can we rename `endTransactionTime` to `exlusiveEndTransactionTime` (and optionally `beginTransactionTime` to `inclusiveBeginTransactionTime)